src/main/java/org/netcracker/educationcenter/elasticsearch/database/operations/ElasticsearchOperations.java

```diff
@@ -0,0 +1,137 @@
+package org.netcracker.educationcenter.elasticsearch.database.operations;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.delete.DeleteRequest;
+import org.elasticsearch.action.delete.DeleteResponse;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.update.UpdateRequest;
+import org.elasticsearch.action.update.UpdateResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.netcracker.educationcenter.elasticsearch.Connection;
+
+import java.io.IOException;
+
+/**
+ * Abstract class for Elasticsearch Database operations (Insert, Get, Delete, Update)
+ *
+ * @author Mikhail Savin
+ */
+public abstract class ElasticsearchOperations {
+    private static final Logger LOG = LogManager.getLogger();
+
+    /**
+     * JSON object mapper
+     */
+    private ObjectMapper mapper;
+
+    /**
+     * Current connection instance
+     */
+    private final Connection connection;
+
+    /**
+     * Constructor with given connection to interact with ES DB.
+     *
+     * @param connection current connection
+     */
+    public ElasticsearchOperations(Connection connection) {
+        this.connection = connection;
+        this.mapper = new ObjectMapper();
+    }
+
+    /**
+     * @return JSON object mapper
+     */
+    public ObjectMapper getMapper() {
+        return mapper;
+    }
+
+    /**
+     * @param mapper JSON object mapper to set
+     */
+    public void setMapper(ObjectMapper mapper) {
```

Comment:
Why need public getter and setter for object mapper?

Reply:
I have deleted setter for object mapper. We need it's getter in subclasses, for instance:
```
 public void insertJiraIssue(JiraIssue jiraIssue) {
        try {
            String jsonString = getMapper().writeValueAsString(jiraIssue);
            insert(jsonString, jiraIssue.getId(), INDEX);
        } catch (JsonProcessingException e) {
            LOG.error(e);
        }
    }
```